pkg/controller/client.go

```diff
@@ -14,6 +14,7 @@ func (c *Controller) PatchStatus(ctx context.Context, patch *v1alpha1.SpiceDBClu
 	for _, c := range patch.Status.Conditions {
 		c.ObservedGeneration = patch.Generation
 	}
+	patch.ManagedFields = nil
```

Comment:
I meant to put this into a separate commit for clarity (and I will update to do that)

The mechanical reason is just that `apply` fails if you have managed fields set. The apiserver uses the value send as the "field manager" to set the managed field values for you, and doesn't let you directly overwrite them.

But the reason this change is in this PR at all is because I combined the `CtxCluster` object and the `CtxClusterStatus` object into one. 

The original idea was that `CtxCluster` was a reference to the `SpiceDBCluster` in the informer cache (which shouldn't be modified) so any handler could recover the "original" object being synced if it needed it. And the `CtxClusterStatus` was a copy of the `SpiceDBCluster` object that would be continually updated if we made changes to it during a sync, so that subsequent handlers would get the "new" view of the object.

But in practice I thought it was hard to remember the distinction, and I just changed `CtxCluster` to be a deepcopy of the object from the cache, and it's easy enough to query the cache directly if we need to recover the original object in the future.